.github/workflows/swift-release.yml

```diff
@@ -0,0 +1,103 @@
+name: Swift Package Release
+
+on:
+  workflow_call:
+    inputs:
+      working-directory:
+        description: "Directory where Swift package is located"
+        required: false
+        type: string
+        default: "."
+      release-tag-prefix:
+        description: "Prefix for release tags"
+        required: false
+        type: string
+        default: "v"
+    secrets:
+      GITHUB_TOKEN:
+        required: true
+
+jobs:
+  release-check:
+    name: Check if version is published
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+        working-directory: ${{ inputs.working-directory }}
+    outputs:
+      current-version: ${{ steps.check.outputs.current-version }}
+      published: ${{ steps.check.outputs.published }}
+      previous-version: ${{ steps.check.outputs.previous-version }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # fetch tags as well
+
+      - name: Read version from Package.Version
```

Comment:
Let's have the `.version` be lowercase since file extensions are typically all lower-case